dll/win32/shell32/dialogs/view.cpp

```diff
@@ -824,7 +824,7 @@ ScanAdvancedSettings(SHELLSTATE *pSS, DWORD *pdwMask)
         }
         if (lstrcmpiW(pEntry->szKeyName, L"SHOWALL") == 0)
         {
-            pSS->fShowAllObjects = !bChecked ? 1 : 0;
+            pSS->fShowAllObjects = bChecked ? 1 : 0;
```

Comment:
```suggestion
            pSS->fShowAllObjects = !!bChecked;
```

Reply:
This is more confusing.

 - All the other code in this function uses the ? : syntax.
 - As you can see, the existing logic was incorrect, don't make it harder to reason about again.

Reply:
Using `!!` to cast towards bool is an absolutely common pattern. It is exclusively used when people want to cast to bool. That is not confusing. Quite the opposite: It is very speaking and clear.
Contrary the `?` operator-style is much harder to read, it carries much more mental load, and is very prone to typing errors.
Hermes suggestion is a good one.

Reply:
I used to think this too, but it is wrong. `!!` is often used in bitwise functions to collapse a result into either 1 or 0.
In fact, I'd avoid using 1 and 0 literals whenever possible.

Reply:
It is not about it being a bit field, it is about `bChecked` vs UI and registry values being "Hide..." and SHELLSTATE using `fShow...` that makes things burdensome to reason about. I don't want to blindly assign the value of `bChecked` to the bit field.